test/test_transitions.py

```diff
@@ -0,0 +1,154 @@
+    def participant(self, exp, a):
+        participant = a.participant()
+        participant.current_item = None
+        exp.grid.players[participant.id] = participant
+        return participant
+
+    def test_pickup_item_success(self, item, exp, participant):
+        exp.handle_item_pick_up(msg={"player_id": participant.id, "position": (0, 0)})
+        assert not exp.grid.item_locations  # The item is no longer there
+        assert participant.current_item == item  # The item is now in the player's inventory
+
+    def test_pickup_item_full_hands_error(self, item, mocked_exp, participant):
+        participant.current_item = self.create_item()
+
+        mocked_exp.handle_item_pick_up(msg={"player_id": participant.id, "position": (0, 0)})
+        assert list(mocked_exp.grid.item_locations.keys()) == [(0, 0)]  # The item is still there
+        assert participant.current_item != item  # The item is not in the player's inventory
+        assert len(self.messages) == 1
+
+    def test_pickup_item_no_item_error(self, item, mocked_exp, participant):
+        mocked_exp.handle_item_pick_up(msg={"player_id": participant.id, "position": (0, 1)})
+        assert list(mocked_exp.grid.item_locations.keys()) == [(0, 0)]  # The item is still there
+        assert participant.current_item != item  # The item is not in the player's inventory
+        assert len(self.messages) == 1
+
+    def test_drop_item_success(self, exp, participant):
+        participant.current_item = item = self.create_item()
+
+        exp.handle_item_drop(msg={"player_id": participant.id, "position": (3, 3)})
+        assert list(exp.grid.item_locations.keys()) == [(3, 3)]
+        assert item.position == (3, 3)
+
+    def test_drop_item_cant_drop_error(self, mocked_exp, participant, item):
+        """The user tries to drop an item over an exiting item.
+        """
+        assert item  # The `item` fixture places it at (0, 0)
+        participant.current_item = self.create_item()  # The participant has an item in their hands
+
+        mocked_exp.handle_item_drop(msg={"player_id": participant.id, "position": (0, 0)})
+        # Dropping failed, so the item is still in the player's hands
+        assert participant.current_item
+        assert len(self.messages) == 1  # and we get an error message
+
+
+class TestHandleItemTransition(object):
+    messages = []
+
+    @pytest.fixture(scope="function")
+    def mocked_exp(self, exp):
+        def publish(error_msg):
+            self.messages.append(error_msg)
+        exp.publish = publish
+        exp.transition_config = {
+            ('last', 2, 3): {
+                'actor_end': 5,
+                'actor_start': 2,
+                'last_use': False,
+                'modify_uses': [0, 0],
+                'target_end': 3,
+                'target_start': 3,
+                'visible': 'always'
+            },
+        }
+        yield exp
+        self.messages[:] = []
+
+    @pytest.fixture
+    def item(self, exp):
```

Comment:
Since you're using this fixture twice, maybe we move it out to the module level (or into `conftest.py`)?